cmd/mqtt/main.go

```diff
@@ -185,7 +225,10 @@ func main() {
 		go chc.CallHome(ctx)
 	}
 
-	var interceptor session.Interceptor
+	interceptor := interceptor{
+		username: cfg.MQTTTargetUsername,
+		password: cfg.MQTTTargetPassword,
+	}
```

Comment:
Why do we need this interceptor to set username and password?

Reply:
RabbitMQ runs in two modes, with and without authentication. For our case devices, i.e, clients always have authentication (thingID and thingKey). So we either add the RabbitMQ username and password when sending the request or remove the username and password when forwarding the request to RabbitMQ

Reply:
In that case, let's remove Auth on RabbitMQ, since we will be using mGate for that anyways.

Reply:
Since authN can't be turned off (i.e. we need at least the default user), let's keep this approach (see [https://www.rabbitmq.com/docs/mqtt#anonymous-connections](https://www.rabbitmq.com/docs/mqtt#anonymous-connections))